plugin/tower/pkg/controller/policy/controller.go

```diff
@@ -591,6 +764,60 @@ func (c *Controller) applyPoliciesChanges(oldKeys []string, new []v1alpha1.Secur
 	return nil
 }
 
+// parseControllerPolicy convert schema.EverouteCluster to []v1alpha1.SecurityPolicy
+func (c *Controller) parseControllerPolicy(clusters []*schema.EverouteCluster) ([]v1alpha1.SecurityPolicy, error) {
+	sp := v1alpha1.SecurityPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ControllerPolicyName,
+			Namespace: c.namespace,
+		},
+		Spec: v1alpha1.SecurityPolicySpec{
+			Tier:        constants.Tier2,
+			DefaultRule: v1alpha1.DefaultRuleNone,
+			PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress, networkingv1.PolicyTypeEgress},
+		},
+	}
+	for _, cluster := range clusters {
+		for _, ctrl := range cluster.ControllerInstances {
+			epName := endpoint.GetCtrlEndpointName(cluster.ID, ctrl)
+			sp.Spec.AppliedTo = append(sp.Spec.AppliedTo, v1alpha1.ApplyToPeer{
+				Endpoint: &epName,
+			})
+		}
+	}
+	if len(sp.Spec.AppliedTo) == 0 {
+		return nil, nil
+	}
+
+	return []v1alpha1.SecurityPolicy{sp}, nil
+}
+
+// parseSystemEndpointsPolicy convert schema.SystemEndpoints to []v1alpha1.SecurityPolicy
+func (c *Controller) parseSystemEndpointsPolicy(systemEndpoints *schema.SystemEndpoints) ([]v1alpha1.SecurityPolicy, error) {
+	sp := v1alpha1.SecurityPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      SystemEndpointsPolicyName,
+			Namespace: c.namespace,
+		},
+		Spec: v1alpha1.SecurityPolicySpec{
+			Tier:        constants.Tier2,
+			DefaultRule: v1alpha1.DefaultRuleNone,
+			PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress, networkingv1.PolicyTypeEgress},
+		},
+	}
+	for _, ip := range systemEndpoints.IPPortEndpoints {
+		epName := endpoint.GetSystemEndpointName(ip.Key)
+		sp.Spec.AppliedTo = append(sp.Spec.AppliedTo, v1alpha1.ApplyToPeer{
+			Endpoint: &epName,
+		})
+	}
+	if len(sp.Spec.AppliedTo) == 0 {
+		return nil, nil
+	}
+
+	return []v1alpha1.SecurityPolicy{sp}, nil
+}
```

Comment:
These two functions can be combined with one to reduce the duplicate code.

Reply:
L780-L787 and L808-L813 have different logic, L769-L779 and L797-L807 do have duplicate code.
Too many duplicate code in this file, such as syncXXXworker, an extra PR will reduce these code.

Reply:
OK